packages/astro/src/core/app/index.ts

```diff
@@ -145,7 +151,23 @@ export class App {
 		return routeData;
 	}
 
-	async render(request: Request, routeData?: RouteData, locals?: object): Promise<Response> {
+	async render(request: Request, options?: RenderOptions): Promise<Response>
+	async render(request: Request, routeData?: RouteData, locals?: object): Promise<Response>
```

Comment:
Adding a `/** @deprecated See https://github.com/withastro/astro/pull/9199 */` comment above the old signature would be good so that users get a hint in their editor.

Reply:
You can deprecate an overload?!